src/pages/home/ReportScreen.js

```diff
@@ -172,6 +185,11 @@ class ReportScreen extends React.Component {
             reportClosedAction = lodashFindLast(this.props.reportActions, action => action.actionName === CONST.REPORT.ACTIONS.TYPE.CLOSED);
         }
 
+        // If a reportActionID exists in the URL then let's close the drawer so that we can see the message comments on native.
+        if (this.getReportActionID()) {
+            Navigation.closeDrawer();
+        }
```

Comment:
>by default the drawer will close

This seems like a strange behavior for a couple of reasons:

1. If we are navigating to the same route I would expect maybe that nothing happens because we don't need to navigate anywhere.
2. If we are navigating to a report route I'd expect the drawer to be open by default not closed and require us to reopen it.

Definitely open to other opinions, but seems preferable to get to the bottom of that instead of adding this logic to the `render()` method and continually forcing the drawer to be closed.

Reply:
So the behavior i've noticed is that even if you click the same link i.e., same route then `BaseDrawerNavigator` re-renders. Once BaseDrawerNavigator re-renders it has a defaultStatus [here](https://github.com/Expensify/App/blob/72222cac403b3d6deab30633bffdfddb1c8b767d/src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js#L57) which for mobile is set as 'open', so whenever it rerender, even for the same route it will open the drawer from my understanding.

We could place `shouldComponentUpdate` in BaseDrawerNavigator and not update it when props.screens is unchanged perhaps? Does that sound right and do you want me to tweak `BaseDrawerNavigator` here?